frontend/ui/quickstart.lua

```diff
@@ -63,16 +84,88 @@ local quickstart_guide = T(_([[
 ---
 You can access the complete user manual from [our GitHub page](https://github.com/koreader/koreader).
 ]])
+end
 ..
-_([[## User interface <a id="ui"></a>
+if Device:hasScreenKB() then
+    _([[## User interface <a id="ui"></a>
```

Comment:
Anyway, I think all your stuff is messed up, you can't just do that, have `..` (strings concatenation) intermixed with `if then`.
You can only intermix `[[text].. (Device:hasScreenKB() and [[kbtext]] or [[touchtext]]) .. `

Reply:
which is why I didn't want to mess with this like this. the three guides was working just fine.

Reply:
is this valid?

```
(Device:hasScreenKB() and [[kbtext]] or Device:hasSymmKey() and [[kb2text]] or [[touchtext]])
```

Reply:
Yes, but presumably the end result would be quite a bit more legible with `table.insert`.

Reply:
Probably not, it would probably need more parenthesis.
But it's also quite unreadable/unfollowable.
If you would follow up on this (not sure you should), I guess that instead of having a single long text (made out concatenated chunks), you should build it incrementally:
```lua
local quickstart = [[common start]
if iskindle3 then
  quickstart = quickstart .. [[k3 stuff]]
else
  quickstart = quickstart .. [[touch stuff]]
end
quickstart = quickstart .. [[more common stuff]]
...
quickstart = quickstart .. T([[ending stuff %1]], rev)
```

Reply:
```lua
local quick_guide = {}
if x then
    table.insert(quick_guide, blah blah blah)
else
    table.insert(...)
end
if x or y then
    table.insert...
else
    table....
end
```

that what you mea?

Reply:
> Probably not, it would probably need more parenthesis.

Oh, I didn't mean if it would or wouldn't work as written, just that the general principle of `"hello" .. (true and "there" or "nobody")` would work.

> that what you mean?

Yes. Then at the end you would do `table.concat()` to merge it all together.

Reply:
Went with poire's approach, sorrry! I am so sorry i used three r's.

Reply:
We don't want it that way because it's a bad precedent. Explanation here: https://www.lua.org/pil/11.6.html

Reply:
(I don't really mind it with strings concatenation, it's not a hot codepath - and we already have some precedents here and there.
Using table.insert() would add some parens around the already heavy `table.insert(zzz, T(_(...),yy))`
)

Reply:
Very well. I hope it won't come back to bite.

Reply:
no more "bad code", I am Team Frenzie on this one now. Sorrry poire. (note the three r's)

Reply:
What concerns me is that "it's fine here" has a tendency to turn into "someone less familiar with Lua/coding and/or KOReader took it as an example to follow." But for this particular example it shouldn't matter too much since it's easy to spot and easy to change.